arithmetization/src/main/java/net/consensys/linea/zktracer/runtime/stack/Stack.java

```diff
@@ -328,95 +310,105 @@ private void create(MessageFrame frame, StackContext pending) {
     {
       final Bytes val4 = getStack(frame, 0);
 
       pending.addArmingLine(
-          new IndexedStackOperation(
-              3, StackOperation.pop(this.height, val4, stackStampWithOffset(0))),
-          new IndexedStackOperation(
-              4, StackOperation.push(this.height - 2, stackStampWithOffset(4))));
+          new IndexedStackOperation(3, StackOperation.pop(height, val4, stackStampWithOffset(0))),
+          new IndexedStackOperation(4, StackOperation.push(height - 2, stackStampWithOffset(4))));
     }
   }
 
   /**
    * @return true if no stack exception has been raised
    */
   public boolean isOk() {
-    return this.status == Status.NORMAL;
+    return status == Status.NORMAL;
   }
 
   /**
    * @return true if a stack underflow exception has been raised
    */
   public boolean isUnderflow() {
-    return this.status == Status.UNDERFLOW;
+    return status == Status.UNDERFLOW;
   }
 
   /**
    * @return true if a stack underflow exception has been raised
    */
   public boolean isOverflow() {
-    return this.status == Status.OVERFLOW;
+    return status == Status.OVERFLOW;
   }
 
   public void processInstruction(final Hub hub, MessageFrame frame, int stackStamp) {
     final CallFrame callFrame = hub.currentFrame();
-    this.stamp = stackStamp;
-    this.height = this.heightNew;
-    this.currentOpcodeData = hub.opCodeData();
-    callFrame.pending(new StackContext(this.currentOpcodeData.mnemonic()));
+    stamp = stackStamp;
+    currentOpcodeData = hub.opCodeData();
+    callFrame.pending(new StackContext(currentOpcodeData.mnemonic()));
 
-    final int alpha = this.currentOpcodeData.stackSettings().alpha();
-    final int delta = this.currentOpcodeData.stackSettings().delta();
+    final int alpha = currentOpcodeData.stackSettings().alpha();
+    final int delta = currentOpcodeData.stackSettings().delta();
 
-    this.heightNew += this.currentOpcodeData.stackSettings().nbAdded();
-    this.heightNew -= this.currentOpcodeData.stackSettings().nbRemoved();
+    Preconditions.checkArgument(heightNew == frame.stackSize());
+    height = frame.stackSize();
+    heightNew += currentOpcodeData.stackSettings().nbAdded();
+    heightNew -= currentOpcodeData.stackSettings().nbRemoved();
 
     if (frame.stackSize() < delta) { // Testing for underflow
-      this.status = Status.UNDERFLOW;
-    } else if (this.heightNew > MAX_STACK_SIZE) { // Testing for overflow
-      this.status = Status.OVERFLOW;
+      status = Status.UNDERFLOW;
+    } else if (heightNew > MAX_STACK_SIZE) { // Testing for overflow
+      status = Status.OVERFLOW;
     }
 
-    // CALL WCP for the SUX/SOX lookup
-    hub.wcp().callLT(this.height, delta);
+    // stack underflow checks happen for every opcode
+    final StackHeightCheck checkForUnderflow = new StackHeightCheck(height, delta);
+    final boolean isNewCheckForStackUnderflow =
+        hub.transients().conflation().stackHeightChecksForStackUnderflows().add(checkForUnderflow);
```

Comment:
We can expose a get method on the hub level to access `stackHeightChecksForStackUnderflows` data structure. The method in `Hub.java` would return `transients().conflation().stackHeightChecksForStackUnderflows()`, if we know that `transients()` and` conflation()` will not return null. We can add checks to avoid `NullPointerExceptions`